### PR TITLE
Implement Organization.invite_user

### LINKED ISSUE #851: Inviting a user via email address using pygithub.
Hi Team,
Have gone through the documentation and found solutions to invite a nameduser. I have use case where i wanted to invite an external user using his E-mail address. But could not see an option to do the same using pygithub where only nameduser is supported.

Kindly help.

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
Utilize the API preview for the organization invite system to allow
organization owners to invite outside colloborators either via passing
in the user directly, or their email address.
    
Fixes #851

### CODE REVIEW

File: github/Organization.py [hunk 1]
```diff
@@ -776,6 +777,32 @@ def get_teams(self):
...
+        """
+        :calls: `POST /orgs/:org/invitations <http://developer.github.com/v3/orgs/members>`_
+        :param email: string
+        :param role: string
+        :param teams: array of :class:`github.Team.Team`
+        :rtype: None
+        """
+        assert isinstance(email, (str, unicode)), email
+        parameters = {
+            'email': email,
```
[the remark this answers is not in the document]
Comment (s-t-e-v-e-n-k) [reply github/Organization.py]: It just strikes me as awkward -- then we have both user and email defaulting to NotSet and then some asserts to make sure at least one of them is set. But I can add it if you wish.

File: github/Organization.py [hunk 2]
```diff
@@ -776,6 +777,32 @@ def get_teams(self):
...
+            assert isinstance(role, (str, unicode)), role
+            assert role in ['admin', 'direct_member', 'billing_manager']
+            parameters["role"] = role
+        if teams is not github.GithubObject.NotSet:
+            assert all(isinstance(team, github.Team.Team) for team in teams)
+            parameters["team_ids"] = [t.id for t in teams]
+        headers, data = self._requester.requestJsonAndCheck(
+            "POST",
+            self.url + "/invitations",
+            headers={'Accept': 'application/vnd.github.dazzler-preview+json'},
```
[the remark this answers is not in the document]
Comment (s-t-e-v-e-n-k) [reply github/Organization.py]: Wow, nice! I didn't notice this cleanup you did, it looks great!

File: github/Organization.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfdye) [reply github/Organization.py]: You can use something like `assert (email is NotSet) ^ (invitee_id is NotSet), "only email or only invitee_id is allowed"`

File: github/Organization.py [hunk 3]
```diff
@@ -776,6 +778,37 @@ def get_teams(self):
...
+        :calls: `POST /orgs/:org/invitations <http://developer.github.com/v3/orgs/members>`_
+        :param user: :class:`github.NamedUser.NamedUser`
+        :param email: string
+        :param role: string
+        :param teams: array of :class:`github.Team.Team`
+        :rtype: None
+        """
+        assert user is github.GithubObject.NotSet or isinstance(user, github.NamedUser.NamedUser), user
+        assert email is github.GithubObject.NotSet or isinstance(email, (str, unicode)), email
+        assert (email is github.GithubObject.NotSet) or (user is github.GithubObject.NotSet), "specify only one of email or user"
```
Comment (sfdye) [inline github/Organization.py]: It seems when both user and email are `NotSet`, these checks will pass without raising an exception? But at least one of them is required.

**Comment by s-t-e-v-e-n-k** [comment]
@sfdye Fixed, since I should have read what you wrote more carefully and used XOR to begin with. I've added a test case too.

**Comment by sfdye** [comment]
@s-t-e-v-e-n-k Awesome!

**Pull request merged**